handle empty lineage entries gracefully

### Steps to reproduce

- have a lineage entry that is empty in either the branch or parent name
- run any Git Town command

expected behavior: it deletes the invalid lineage entries

actual behavior: it crashes

### Command output

```shell
panic: local branch names cannot be empty

goroutine 1 [running]:
github.com/git-town/git-town/v14/src/git/gitdomain.NewLocalBranchName(...)
        C:/Users/kevin/d/git-town/src/git/gitdomain/local_branch_name.go:16
github.com/git-town/git-town/v14/src/git/gitdomain.NewLocalBranchNames(...)
        C:/Users/kevin/d/git-town/src/git/gitdomain/local_branch_names.go:16
github.com/git-town/git-town/v14/src/git/gitdomain.ParseLocalBranchNames({0xc000268168?, 0xbc00000001756ca0?})
        C:/Users/kevin/d/git-town/src/git/gitdomain/local_branch_names.go:28 +0xc5
github.com/git-town/git-town/v14/src/config/gitconfig.AddKeyToPartialConfig({0x166ccdb, 0x1b}, {0xc000268168, 0xe}, 0xc000114f30)
        C:/Users/kevin/d/git-town/src/config/gitconfig/access.go:103 +0xe1f
github.com/git-town/git-town/v14/src/config/gitconfig.(*Access).load(_, _, _)
        C:/Users/kevin/d/git-town/src/config/gitconfig/access.go:256 +0x553
github.com/git-town/git-town/v14/src/config/gitconfig.(*Access).LoadLocal(...)
        C:/Users/kevin/d/git-town/src/config/gitconfig/access.go:33
github.com/git-town/git-town/v14/src/execute.OpenRepo({0x0, 0x0, 0x1, 0x1, 0x0, 0x0})
        C:/Users/kevin/d/git-town/src/execute/open_repo.go:47 +0x307
github.com/git-town/git-town/v14/src/cmd.executeSwitch(0x0, 0x0?)
        C:/Users/kevin/d/git-town/src/cmd/switch.go:38 +0x5f
github.com/git-town/git-town/v14/src/cmd.switchCmd.func1(0xc0001cc800?, {0x1cf2260?, 0x4?, 0x165a9a9?})
        C:/Users/kevin/d/git-town/src/cmd/switch.go:29 +0x47
github.com/spf13/cobra.(*Command).execute(0xc0001bdb00, {0x1cf2260, 0x0, 0x0})
        C:/Users/kevin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
        C:/Users/kevin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        C:/Users/kevin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/git-town/git-town/v14/src/cmd.Execute()
        C:/Users/kevin/d/git-town/src/cmd/core.go:38 +0x619
```


### Additional information

_No response_

Repo 1:

```
> git config --add git-town-branch..parent main                                                                                     

> git town config

panic: local branch names cannot be empty

goroutine 1 [running]:
github.com/git-town/git-town/v14/src/git/gitdomain.NewLocalBranchName(...)
	/Users/kevlar/d/git-town/src/git/gitdomain/local_branch_name.go:12
github.com/git-town/git-town/v14/src/config/gitconfig.AddKeyToPartialConfig({0x140001d064e, 0x17}, {0x140001d0666, 0x4}, 0x14000174150)
	/Users/kevlar/d/git-town/src/config/gitconfig/access.go:41 +0x135c
github.com/git-town/git-town/v14/src/config/gitconfig.(*Access).load(_, _, _)
	/Users/kevlar/d/git-town/src/config/gitconfig/access.go:261 +0x48c
github.com/git-town/git-town/v14/src/config/gitconfig.(*Access).LoadLocal(...)
	/Users/kevlar/d/git-town/src/config/gitconfig/access.go:33
github.com/git-town/git-town/v14/src/execute.OpenRepo({_, _, _, _, _, _})
	/Users/kevlar/d/git-town/src/execute/open_repo.go:49 +0x274
github.com/git-town/git-town/v14/src/cmd/config.executeDisplayConfig(0x0?)
	/Users/kevlar/d/git-town/src/cmd/config/root.go:36 +0x70
github.com/git-town/git-town/v14/src/cmd/config.RootCmd.func1(0x1400020a900?, {0x103d7d6a0?, 0x4?, 0x1034a6128?})
	/Users/kevlar/d/git-town/src/cmd/config/root.go:26 +0x28
github.com/spf13/cobra.(*Command).execute(0x14000127200, {0x103d7d6a0, 0x0, 0x0})
	/Users/kevlar/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0x82c
github.com/spf13/cobra.(*Command).ExecuteC(0x14000126300)
	/Users/kevlar/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/kevlar/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/git-town/git-town/v14/src/cmd.Execute()
	/Users/kevlar/d/git-town/src/cmd/core.go:38 +0x3a8
main.main()
	/Users/kevlar/d/git-town/main.go:18 +0x28

```

--- a/src/config/gitconfig/access.go
+++ b/src/config/gitconfig/access.go
@@ -30,11 +30,13 @@ func (self *Access) AddKeyToPartialConfig(key Key, value string, config *configd
 		}
 		childName := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(key.String(), LineageKeyPrefix), LineageKeySuffix))
 		if childName == "" {
+			// empty lineage entries are invalid --> delete it
 			return self.RemoveLocalConfigValue(key)
 		}
 		child := gitdomain.NewLocalBranchName(childName)
 		value = strings.TrimSpace(value)
 		if value == "" {
+			// empty lineage entries are invalid --> delete it
 			return self.RemoveLocalConfigValue(key)
 		}
 		parent := gitdomain.NewLocalBranchName(value)

--- a/src/config/gitconfig/access.go
+++ b/src/config/gitconfig/access.go
@@ -33,6 +33,10 @@ func (self *Access) AddKeyToPartialConfig(key Key, value string, config *configd
 			return self.RemoveLocalConfigValue(key)
 		}
 		child := gitdomain.NewLocalBranchName(childName)
+		value = strings.TrimSpace(value)
+		if value == "" {
+			return self.RemoveLocalConfigValue(key)
+		}
 		parent := gitdomain.NewLocalBranchName(value)
 		config.Lineage[child] = parent
 		return nil

